File: test_shopee_all/3.py
import pandas as pd

df = pd.read_csv('test_shopee_all/Orders.csv')
df['Order Date'] = pd.to_datetime(df['Order Date'])
print(df)

hasil = df.groupby('Customer ID')['Order Date'].min().reset_index(name = 'pembelian_pertama').sort_values('Customer ID', ascending=True).reset_index(drop=True)
print(hasil)

File: test_shopee_all/3.r
library(dplyr)
library(lubridate)

df <- read.csv('test_shopee_all/Orders.csv')
# df$Order.Date <- mdy(df$Order.Date)
df$Order.Date <- as.Date(df$Order.Date, format = '%m/%d/%Y')
print(head(df))

hasil <- df %>% group_by(Customer.ID) %>% summarise(pembelian_awal = min(Order.Date)) %>% arrange(Customer.ID)
print(head(hasil))
print(tail(hasil))